batcher/aligned-task-sender/src/commands.rs

```diff
@@ -0,0 +1,453 @@
+    std::fs::create_dir_all(args.dir_to_save_proofs.clone()).expect("Could not create directory");
+
+    let mut handles = vec![];
+    for i in 1..args.number_of_proofs + 1 {
+        let dir_to_save_proofs = args.dir_to_save_proofs.clone();
+
+        let handle = thread::spawn(move || {
+            match args.proof_type {
+                ProofType::Groth16 => {
+                    let dir_to_save_proofs =
+                        format!("{}/groth16_{}/", dir_to_save_proofs.clone(), i);
+
+                    // we need to create the directory as the go script does not handle it
+                    if let Err(e) = fs::create_dir(dir_to_save_proofs.clone()) {
+                        if e.kind() != ErrorKind::AlreadyExists {
+                            eprintln!("Error creating directory: {}", e);
+                            // Handle or log the error, but don't panic.
+                        }
+                    }
+
+                    Command::new("go")
+                        .arg("run")
+                        .arg(GROTH_16_PROOF_GENERATOR_FILE_PATH)
+                        .arg(format!("{:?}", i))
+                        .arg(dir_to_save_proofs)
+                        .status()
+                        .unwrap();
+                }
+            }
+        });
+        handles.push(handle);
+    }
+
+    for handle in handles {
+        let _ = handle.join();
+    }
+}
+
+pub async fn generate_and_fund_wallets(args: GenerateAndFundWalletsArgs) {
+    if matches!(args.network.into(), Network::Devnet) {
+        let Ok(eth_rpc_provider) = Provider::<Http>::try_from(args.eth_rpc_url.clone()) else {
+            error!("Could not connect to eth rpc");
+            return;
+        };
+        let Ok(chain_id) = eth_rpc_provider.get_chainid().await else {
+            error!("Could not get chain id");
+            return;
+        };
+        let amount_to_deposit_to_aligned =
+            parse_ether(&args.amount_to_deposit_to_aligned).expect("Ether format should be: XX.XX");
+
+        let file = match File::open(&args.private_keys_filepath) {
+            Ok(f) => f,
+            Err(err) => {
+                error!("Could not open private keys file: {}", err.to_string());
+                return;
+            }
+        };
+        let file_reader = BufReader::new(file);
+
+        let mut last_handle = None;
+        for (i, line) in file_reader.lines().enumerate() {
+            // Load the wallet
+            let private_key_str = line.unwrap();
+            let wallet = Wallet::from_str(private_key_str.trim())
+                .expect("Invalid private key")
+                .with_chain_id(chain_id.as_u64());
+
+            // Send funds to aligned from the wallet
+            let funded_wallet_signer =
+                SignerMiddleware::new(eth_rpc_provider.clone(), wallet.clone());
+            tokio::time::sleep(Duration::from_millis(50)).await; // To avoid overloading the RPC
```

Comment:
nit: This may be a bottleneck, we could try to improve it in the future if it takes too long:
* We could send 3 transactions and then wait for ~1 min
* We could use different RPC nodes

Reply:
Unless it actually becomes a bottleneck, it's not a bottleneck, but if we try something and it stops here we check

Reply:
this is not really a problem, funding 1000 wallets takes less than a minute. And it is necesarry the small await for not overloading the RPC (devnet, so we cant have many, since it is local).

Also, the suggested fix does 3 transactions every 1 minute. this one does more than 1000 transactions per minute